Trunk compilation is broken with the Swift 5.7 compiler on linux

I [reported this earlier on the offending pull](https://github.com/apple/swift-system/pull/116#discussion_r1056176682), but filing an issue to make sure it is handled.

The error:
```
Sources/System/MachPort.swift:75:5: error: deinitializers may only be declared within a class or actor
    deinit {
    ^
```

Yes, I'm hitting this too with 5.7 releases.

```
$ git rev-parse HEAD
ccea4eed1aba98022a7af65875ffdf2d821450da
$ jw-docker-swift-5.7  swift build
Building for debugging...
/Users/johannes/devel/swift-system/Sources/System/MachPort.swift:75:5: error: deinitializers may only be declared within a class or actor
    deinit {
    ^
/Users/johannes/devel/swift-system/Sources/System/MachPort.swift:75:5: error: deinitializers may only be declared within a class or actor
    deinit {
    ^
/Users/johannes/devel/swift-system/Sources/System/MachPort.swift:75:5: error: deinitializers may only be declared within a class or actor
    deinit {
    ^
[25/25] Compiling SystemPackage Exports.swift
```

--- a/Sources/System/MachPort.swift
+++ b/Sources/System/MachPort.swift
@@ -7,7 +7,7 @@
  See https://swift.org/LICENSE.txt for license information
 */
 
-#if $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
+#if swift(>=5.8) && $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
 
 import Darwin.Mach
 

--- a/Tests/SystemTests/MachPortTests.swift
+++ b/Tests/SystemTests/MachPortTests.swift
@@ -7,7 +7,7 @@
  See https://swift.org/LICENSE.txt for license information
 */
 
-#if $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
+#if swift(>=5.8) && $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
 
 import XCTest
 import Darwin.Mach

